app/scripts/components/exploration/hooks/use-analysis-data-request.ts

```diff
@@ -107,23 +114,39 @@ export function useAnalysisDataRequest({
     };
 
     const { start, end } = selectedInterval;
-
-    requestDatasetTimeseriesData({
-      maxItems: MAX_QUERY_NUM,
-      start,
-      end,
-      aoi,
-      dataset,
-      queryClient,
-      concurrencyManager: analysisConcurrencyManager,
-      onProgress: (data) => {
-        setAnalysis(data);
-      }
-    });
+    async function makeCall(){
+      const stat = await requestDatasetTimeseriesData({
+        maxItems: MAX_QUERY_NUM,
+        start,
+        end,
+        aoi,
+        dataset,
+        queryClient,
+        concurrencyManager: analysisConcurrencyManager,
+        onProgress: (data) => {
+          setAnalysis(data);
+        }
+      });
+      setAnalysisResult(stat);
+    }
+    makeCall();
     // We want great control when this effect run which is done by incrementing
     // the analysisRun. This is done when the user refreshes the analysis or
     // when they enter the analysis. It is certain that when this effect runs
     // the other values will be up to date. Adding all dependencies would cause
     // the hook to continuously run.
   }, [analysisRunId, datasetStatus, isAnalyzing]);
+
+
+  useEffect(() => {
+    // setAnalysis sets Jotai Atom value for analysis result
+    // It should sequentially set the analysis progress as 
+    // idle => loading (increasing number of loaded items) => success
+    // However, it is not setting the value sequentially, 
+    // ended up loading status as a result even when the request is successful
+    // This just overwrites the analysis result with the final result of analysis, 
+    // so Jotai value gets 'corrected' in case it is wrong
+    setAnalysis(analysisResult);
+  },[setAnalysis, analysisResult]);
```

Comment:
💭 : now i'm not sure but instead of this useEffect which watches `analysisResult` as a dependency. Could we just call `setAnalysis` on line 130 instead of doing `setAnalysisResult(stat);` and then in that case we wouldn't need this local state?

Reply:
Yeah, that was my thought too first, too, but that approach doesn't set the state stably. (Observing the same pattern of state bouncing back and forth) 

~~ After a while ~~

OH I THINK I found a more concrete clue, I will post the finding in the main thread.